packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -531,6 +514,10 @@ export abstract class BrowserCrawler<
             page as any,
         ) as ProvidedController;
 
+        const session = useIncognitoPages
+            ? crawlingContext.session
+            : (browserControllerInstance.launchContext.session as Session);
```

Comment:
Not directly related to your changes, but is there anything we could do to make this mechanism less confusing? A warning when you start the browser crawler with a non-default session pool and without `useIncognitoPages` for example?

Or do we expect to solve this on a higher level later on?

Reply:
Yes, this already is an improvement (because users will see the session/proxy URL that's actually being used), but I agree we should improve the `BrowserPool` / `SessionPool` integration and make this more transparent later on.